lib/engine-addon.js

```diff
@@ -696,17 +720,20 @@ module.exports = {
         }
 
         // Merge all of our final trees!
-        return mergeTrees(
-          [
-            publicRelocated,
-            addonsEnginesPublicTreesMerged,
-            otherAssets,
-            finalStylesTree,
-            vendorJSImportTree,
-            concatEngineTree
-          ].filter(Boolean),
-          { overwrite: true }
-        );
+        var finalMergeTrees = [
+          publicRelocated,
+          addonsEnginesPublicTreesMerged,
+          otherAssets,
+          finalStylesTree,
+          vendorJSImportTree,
+          concatEngineTree
+        ];
+        if (this.lazyLoading.includeRoutesInApplication === false) {
```

Comment:
Followed your suggestion of refactoring this into a single function that returns the tree to be merged in.